connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigStorage.java

```diff
@@ -396,6 +398,7 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
                 synchronized (lock) {
                     if (value.value() == null) {
                         // Connector deletion will be written as a null value
+                        log.info("Removed connector " + connectorName + " due to null configuration. This is usually intentional and does not indicate an issue.");
```

Comment:
Do you actually want this at info level? This looks like the one case that will add log messages at the default level. Deletions ideally aren't all that frequent, but on worker start up we'll log all the deletions on INFO level as the config log is consumed to catch up.